File: src/main/java/se/kth/spork/exception/SporkException.java
package se.kth.spork.exception;

/**
 * Base exception for Spork exceptions.
 *
 * @author Simon Larsén
 */
public abstract class SporkException extends RuntimeException {
    public SporkException(String s) {
        super(s);
    }

    public SporkException(String s, Throwable throwable) {
        super(s, throwable);
    }
}

File: src/main/kotlin/se/kth/spork/exception/SporkException.kt
package se.kth.spork.exception

import java.lang.RuntimeException

/**
 * Base exception for Spork exceptions.
 *
 * @author Simon Larsén
 */
abstract class SporkException @JvmOverloads constructor(s: String, t: Throwable? = null) : RuntimeException(s, t)
